internal/ingest/linksystem.go

```diff
@@ -451,6 +447,9 @@ func (ing *Ingester) indexContentBlock(adCid cid.Cid, pubID peer.ID, nentries ip
 	//
 	// Do not return here if error; try to index content in this chunk.
 	hasNextLink, linkErr := ing.setNextCidToAd(nchunk, adCid)
+	if linkErr != nil {
+		return linkErr
+	}
```

Comment:
See comment above:
```go
// Do not return here if error; try to index content in this chunk.
```

This was done to process the current entry, even if the link to the next entry could not be read.  Either remove the code that returns, or update the comment.